src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IDynamicMemberReferenceExpression.vb

```diff
@@ -639,6 +640,66 @@ Block[B5] - Exit
 
             VerifyFlowGraphAndDiagnosticsForTest(Of MethodBlockSyntax)(source, expectedFlowGraph, expectedDiagnostics)
         End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <WorkItem(27034, "https://github.com/dotnet/roslyn/issues/27034")>
+        <Fact()>
+        Public Sub DynamicMemberReference_OffObjectCollectionInitializer()
+            Dim source = <![CDATA[
+Imports System
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Runtime.CompilerServices
+
+Module Mod1
+    Class C
+        Implements IEnumerable(Of Integer)
+
+        Sub M(a As Object, b As Object)
+            Dim i = New C From {a}.Add(b)'BIND:"New C From {a}.Add"
```

Comment:
Ensures that the member reference in the explicit add call is actually explicit in the tree.